src/main/webapp/index_script.js

```diff
@@ -13,37 +13,29 @@
 // limitations under the License.
 
 document.addEventListener("DOMContentLoaded", async function() {
-  let organizations = [{
-    "name": "Organization A",
-  },
-  {
-    "name": "Organization B",
-  },
-  {
-    "name": "Organization C",
-  },
-  {
-    "name": "Organization D",
-  },
-  {
-    "name": "Organization E",
-  },
-  {
-    "name": "Organization F"
-  }];
   const organizationArea = document.getElementById('organization-list');
-  organizations.forEach((organization) => {
-    organizationArea.appendChild(createOrganization(organization));
-  });
+  renderOrganizations(organizationArea);
 });
 
+/* 
+ * This async function gets a default list of organization names to display when the dom content
+ * loads by not passing any parameters to /list-organizations
+ */
+async function renderOrganizations(organizationArea) {
+  const response = await fetch(`/list-organizations`);
+  const organizationNames = await response.json();
+  for (let i = 0; i < organizationNames.length; i++) {
+    organizationArea.appendChild(createOrganization(organizationNames[i]));
+  }
+}
```

Comment:
you could make the onload function async and just do the fetch and await in there as well as using a for each loop so its shorter

Reply:
My thoughts in breaking this off to be its own function are that in the future we can just add a single parameter to it that tells it whether or not to load just the user's organizations, and then that parameter can be passed to the servlet. This way, this one function can handle all instances of needing to load a list of organizations to an HTML div, from anywhere in the website, as opposed to just happening when this page is loaded & having to rewrite the code for different situations.

Reply:
okay that makes sense!